src/engine/enginemaster.cpp

```diff
@@ -443,22 +546,79 @@ void EngineMaster::process(const int iBufferSize) {
-        // with the channel gains and overall master gain.
-        if (!m_pMasterTalkoverMix->toBool()) {
-            // Add Talkover to Master output
-            SampleUtil::copy4WithGain(m_pMaster,
-                    m_pOutputBusBuffers[EngineChannel::LEFT], 1.0,
-                    m_pOutputBusBuffers[EngineChannel::CENTER], 1.0,
-                    m_pOutputBusBuffers[EngineChannel::RIGHT], 1.0,
+        // Make master and booth mixes from the
+        // crossfader orientation buffers and talkover mix
+        SampleUtil::copy3WithGain(m_pMaster,
+            m_pOutputBusBuffers[EngineChannel::LEFT], 1.0,
+            m_pOutputBusBuffers[EngineChannel::CENTER], 1.0,
+            m_pOutputBusBuffers[EngineChannel::RIGHT], 1.0,
+            iBufferSize);
+
+        if (configuredTalkoverMixMode != TalkoverMixMode::DIRECT_MONITOR
+            && m_pNumMicsConfigured->get() > 0) {
+              m_pInputLatencyCompensationDelay->process(m_pMaster, iBufferSize);
+        }
+
+        if (boothEnabled) {
+            CSAMPLE boothGain = m_pBoothGain->get();
+
+            if (configuredTalkoverMixMode == TalkoverMixMode::MASTER) {
+                // Mix the talkover mix with the master output, but exclude it from
+                // the booth output.
+                if (m_bRampingGain) {
+                    SampleUtil::copy1WithRampingGain(m_pBooth, m_pMaster,
+                        m_boothGainOld, boothGain, iBufferSize);
+                } else {
+                    SampleUtil::copy1WithGain(m_pBooth, m_pMaster,
+                        boothGain, iBufferSize);
+                }
+
+                SampleUtil::copy2WithGain(m_pMaster,
+                    m_pMaster, 1.0,
+                    m_pTalkover, 1.0,
+                    iBufferSize);
+            } else if (configuredTalkoverMixMode == TalkoverMixMode::MASTER_AND_BOOTH) {
+                // Mix the talkover mix with the master output and include
+                // it in the booth output too.
+                SampleUtil::copy2WithGain(m_pMaster,
+                    m_pMaster, 1.0,
                     m_pTalkover, 1.0,
                     iBufferSize);
+
+                if (m_bRampingGain) {
+                    SampleUtil::copy1WithRampingGain(m_pBooth, m_pMaster,
+                        m_boothGainOld, boothGain, iBufferSize);
+                } else {
+                    SampleUtil::copy1WithGain(m_pBooth, m_pMaster,
+                        boothGain, iBufferSize);
+                }
+            } else if (configuredTalkoverMixMode == TalkoverMixMode::DIRECT_MONITOR) {
+                // Skip mixing the talkover mix with the master mix if using direct monitoring
+                // because it is being mixed in hardware without the latency of sending
+                // the signal in to Mixxx for processing.
+                // The talkover mix will be mixed later with recording/broadcasting signal
+                // after latency compensation.
```

Comment:
Ah OK, Now I see. Unfortunately it does not work, because the master buffer already contains the talkover signal. You need to delay the master signal before mixing the talkover signal in . 

These confusions is the reason why I have original suggested to use internal buffers and only link the external buffers to it. This would make the code easy to read. 
We just have to find nice names for the internal buffers.

For example pMixerOut and pTalkover and pMasterMix, 

In this case (direct monitoring):
pMasterMix = delay(pMixerOut) + pTakover  
pMixerOut -> pMaster
pMixerOut -> pBooth 
pMasterMix -> pSidechain 

master and booth:
pMasterMix = pMixerOut + pTakover  
pMixerMix -> pMaster
pMixerMix -> pBooth 
pMasterMix -> pSidechain 

master only (closed headphones)
pMasterMix = delay(pMixerOut) + pTakover  
pMixerMix -> pMaster
pMixerOut -> pBooth 
pMasterMix -> pSidechain